src/ripple/app/misc/impl/ValidatorSite.cpp

```diff
@@ -213,12 +214,17 @@ ValidatorSite::onTimer (
     std::size_t siteIdx,
     error_code const& ec)
 {
-    if (ec == boost::asio::error::operation_aborted)
-        return;
     if (ec)
     {
-        JLOG(j_.error()) <<
-            "ValidatorSite::onTimer: " << ec.message();
+        // Restart the timer if any errors are encountered, except for the
+        // "operation aborted" error which can be encountered when we cancel
+        // the timer, during shutdown.
+        if (ec != boost::asio::error::operation_aborted)
+        {
+            JLOG(j_.error()) << "ValidatorSite::onTimer: " << ec.message();
+            setTimer();
```

Comment:
I would wait the refresh interval, even on error. If there is some permanent error, we don't want this chewing up resourses.